.github/workflows/ci_template.yml

```diff
@@ -32,7 +32,7 @@ jobs:
         poetry-version: 1.8.3
 
     - name: Install dependencies
-      run: poetry install --with dev
+      run: poetry install
```

Comment:
So this installs optional dependencies by default?

Reply:
Yep. Now, if you do NOT want them, you do, for example, `poetry install --without dev`, but the default position is install all dependencies. As mentioned above, poetry is for developers, not for end users, and they will typically want all the pedendencies.

Reply:
Not exactly. Poetry installs all groups by default except for `extras` and those marked as optional. Previously, the `dev` and `docs` groups were marked as optional, but this PR changes that